runtime/doc/lsp.txt

```diff
@@ -663,6 +672,102 @@ client_is_stopped({client_id})                   *vim.lsp.client_is_stopped()*
+
+                 vim.lsp.config({
+                   autostart = false,
+                   flags = { allow_incremental_sync = true },
+                   servers = {
+                     c = {
+                       autostart = true,
+                       cmd = { "clangd", "--background-index" },
+                       root_markers = { "compile_commands.json", "compile_flags.txt" },
+                     },
+                     lua = {
+                       cmd = { "lua-language-server" },
+                       root_markers = { ".luarc.json", "luarc.json" },
+                       single_file_support = false,
+                     },
+                     go = {
+                       cmd = { "gopls" },
+                       root_markers = { "go.mod" },
+                       settings = {
+                         gopls = {
+                           analyses = {
+                             unusedparams = true,
+                           },
+                         },
+                       },
+                     },
+                   }
+                 })
+<
+
+                Parameters: ~
+                    {config}  (table) Configuration table. Any of the
+                              configuration keys listed in
+                              |vim.lsp.start_client()| can be used here and
+                              will act as the default value for all servers.
+                              The "root_dir" option can also be a function, in
+                              which case it is passed the file path and buffer
+                              number of the buffer and should return the root
+                              directory to use. The following keys are also
+                              recognized:
+                              • autostart (boolean or function, default true):
+                                Automatically start or attach an LSP client
+                                for new buffers. When a function, it is called
+                                with the file name and buffer number of the
+                                buffer and should return a boolean indicating
+                                if the client should autostart. When autostart
+                                is false, a configured LSP server can be
+                                started manually using |vim.lsp.buf_start()|.
+                              • root_markers (table): A list of file or
+                                directory names which indicate the root of a
+                                "project". The default value uses ".git",
+                                ".hg", and ".svn" as root markers, but
+                                typically LSP servers will specify their own
+                                root markers. This option is mutually
+                                exclusive with "root_dir".
+                              • single_file_support (boolean, default true):
```

Comment:
Maybe `false` would be the safer default. Afaik there are servers who otherwise will start indexing everything in your current working directory, which may be $HOME